app/views/components/_navbar.html.erb

```diff
@@ -0,0 +1,43 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light shadow-sm p-3 mb-5 bg-body px-5">
+  <div class="container-fluid">
+    <a class="navbar-brand fw-bold" href="#">Blog App</a>
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarText" aria-controls="navbarText" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarText">
+      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+        <li class="nav-item">
+          <%= link_to "Home", root_path, class: "nav-link", id: 'home' %>
+        </li>
+        <li class="nav-item">
+          <%= link_to "Articles", articles_path, class: "nav-link", id: 'articles' %>
+        </li>
+         <% if current_user %>
+          <% if current_user.admin? || current_user.moderator? %>
```

Comment:
diusahain ga nested if, apalagi klo tumpukannya lgsung kyak gini
`current_user && (current_user.admin? || current_user.moderator?)` atau 
`current_user&.admin? || current_user&.moderator?` ruby ada operator `&.` klo object depannya nil return nya bakal nil, bukan error method of nil object.